Source/Core/VideoBackends/Software/CopyRegion.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "Common/MathUtil.h"
+
+#include <cmath>
+
+namespace SW
+{
+// Modified from
+// http://tech-algorithm.com/articles/nearest-neighbor-image-scaling/
+template <typename T>
+void copy_region(T* const source, const MathUtil::Rectangle<int>& srcrect, T* destination,
```

Comment:
I had meant a pointer to const, not a const pointer (i.e. `const T*`, since the data isn't mutated within the function). Technically, `const T* const` would also be valid here, since the address of the pointer isn't changed either.